src/audio/asrc/asrc.c

```diff
@@ -530,7 +530,8 @@ static int asrc_params(struct comp_dev *dev,
 	sinkb = list_first_item(&dev->bsink_list, struct comp_buffer,
 				source_list);
 
-	sinkb->stream.rate = asrc_get_sink_rate(&cd->ipc_config);
+	if (asrc_get_sink_rate(&cd->ipc_config))
```

Comment:
probably good to state the logic around the IPC value of 0 in a one line comment so its obvious to everyone.